src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -119,6 +119,15 @@ public Socket(AddressFamily addressFamily, SocketType socketType, ProtocolType p
             }
         }
 
+        public Socket(SocketInformation socketInformation)
+        {
+            //
+            // This constructor works in conjunction with DuplicateAndClose, which is not supported.
+            // See comments in DuplicateAndClose.
+            //
+            throw new PlatformNotSupportedException(SR.net_sockets_duplicateandclose_notsupported);
+        }
+
```

Comment:
nit: CoreFx coding guidelines say to not use extra "//" at the top and bottom of comments.  So, this could be instead. The rest of networking code and other libraries currently follow that pattern.

```c#
 // This constructor works in conjunction with DuplicateAndClose, which is not supported.
// See comments in DuplicateAndClose.
```